config/config.py

```diff
@@ -140,22 +108,41 @@ class MinioProperties(BaseSettings):
         secure (bool): Flag indicating whether to use a secure connection.
     """
 
-    class Config:
-        env_prefix = "MINIO_"
+    model_config = ConfigDict(env_prefix="MINIO_") # type: ignore
+
+
+class ConsumerAppConfig(BaseModel):
+    kafka_topic: str
+    max_output_file_size: int
+    flush_interval: int
+    root_path: str
+
+
+class ConsumerConfig(BaseModel):
+    kafka: KafkaConsumerProperties
+    minio: MinioProperties
+    app: ConsumerAppConfig
 
 
 @lru_cache
-def get_config() -> dict[str, dict[str, Any]]:
-    """
-    Retrieve the application configuration, cached for improved performance.
-    Output:
-        dict[str, dict[str, Any]]: A dictionary containing all configuration sections.
-    """
-    CONFIG_FILE_PATH = get_config_path()
-    config_parser = ConfigParser(CONFIG_FILE_PATH)
-    all_configs_dict = config_parser.get_all_configs_dict()
+def get_producer_config() -> ProducerConfig:
```

Comment:
here we just feed raw config dict into the pydantic class, and it parses it recursively. no need to config parser